master/internal/command/notebook_manager.go

```diff
@@ -170,7 +173,18 @@ func (n *notebookManager) newNotebook(req *commandRequest) (*command, error) {
 	taskID := scheduler.NewTaskID()
 
 	// Postprocess the config. Add Jupyter and configuration to the container.
+
+	// Select a random port from the range to assign to the notebook. In host
+	// mode, this mitigates the risk of multiple notebook processes binding
+	// the same port on an agent.
+	port := getPort(minNotebookPort, maxNotebookPort)
+	notebookPorts := map[string]int{"notebook": port}
 	config.Environment.Ports = notebookPorts
+	config.Environment.EnvironmentVariables = model.RuntimeItems{
```

Comment:
question: should we appending to the environment variables instead of overwriting them?

Reply:
Forgot about the user defined ones. Fixed